### DISCUSSION TITLE
Managing Category and Subcategory Relationships: Query String vs. Route Parameters for Passing category_id?

### DISCUSSION THREAD

**Discussion by degodfather** [filamentphp/filament#9343] [Help]
Hello,

I'm very new to FilamentPHP and Livewire, and I'm trying to implement a category and subcategory architecture.

- Category A
  - Subcategory A1
  - Subcategory A2
- Category B
  - Subcategory B1
  - Subcategory B2

Previously, when using Backpack, I would use route parameters to pass the `category_id` for sub-category listing, creation, and editing:

- `admin/{category_id}/sub-category`
- `admin/{category_id}/sub-category/create`
- `admin/{category_id}/sub-category/{sub-category_id}/edit`

However, with Filament, I couldn't find any resources to achieve this. I attempted to use the `tableFilters` query string, but I encountered some issues. Please suggest a solution or share any documentation that can help me achieve this.

**Comment by VincvanAs** [comment]
Maybe this solution will work for you. 

Add nullable parent_id to Category model and define the relation for parent and children (sub) relation:

```
public function children(): HasMany
    {
      return $this->hasMany(Category::class, 'parent_id');
    }

    public function parent(): BelongsTo
    {
        return $this->belongsTo(Category::class, 'parent_id');
    }
```

In your form you could set something like:

```
Forms\Components\Select::make('parent_id')
->label('Main Category')
->disableOptionWhen(fn (string $value, ?Model $record) => (int)$value === (int)$record?->id)
->options(fn() => Category::orderBy('name')->whereNull('parent_id')->pluck('name', 'id')->toArray())
```

Table:

`
TextColumn::make('parent.name')->label('Main Category')->placeholder('-'),
`

Filter:

```
SelectFilter::make('parent_id')
->label('Main Category')
->options(fn() => Category::whereNull('parent_id')->pluck('name', 'id')->toArray()),
```

Let me know if this helps you out, or you looking for anything more specific!

**Reply by degodfather** [reply]
How do I list `children` of a specific `parent` without using a dropdown parent ? I want to use `where` clause in table query. The problem is that I'm not able to pass parent_id from parents listing page to the children listing page. Something similar to the following. 


```
admin/{category_id}/sub-category
admin/{category_id}/sub-category/create
admin/{category_id}/sub-category/{sub-category_id}/edit
```

How do I get the category_id value?

**Reply by VincvanAs** [reply]
If you create a custom page 

https://filamentphp.com/docs/3.x/panels/resources/custom-pages

You can access the selected record by 

`
$this->record->id; // category_id
$this->record->parent_id; // Main catorgory id
$this->record->children; // sub
`

At this point I'm not really sure on what your goal is. Maybe if you share some code, I could assist you better
